src/components/SummaryKeyComponents/TLDRComponent.js

```diff
@@ -0,0 +1,40 @@
+import { getPaperTLDR } from "../../services/TLDRService";
+
+/**
+ * Class representing the TLDR Accordion item
+ * @property {Object} components object that holds DOM elements that represent this component, as well as component's context
+ * @property {HTMLElement} components.tldrText html p element that represents the text of the accordion item
+ */
+class TLDRComponent {
+  components = {
+    tldrText: document.querySelector("#tldrText"),
+  };
+
+  /**
+   * Set the pdf document needed to retrieve the TLDR
+   * Once the pdf is ready, start to fetch the tldr
+   * @param pdfDoc the current pdf document
+   */
+  setPdf = (pdfDoc) => {
```

Comment:
Yeah I agree with both of you, we should maybe publish an event in loadPdf method of PdfReaderComponent class and subscribe all other components to that event instead of using "setters".